go-controller/pkg/libovsdb/util/acl_test.go

```diff
@@ -170,3 +170,131 @@ func TestGetL4Match(t *testing.T) {
 		assert.Equal(t, tc.expected, l4Match)
 	}
 }
+
+func TestGetL3L4MatchesFromNamedPorts(t *testing.T) {
+	testcases := []struct {
+		desc     string
+		protocol []string
+		ports    map[string]*[]NamedNetworkPolicyPort
+		expected []string // per protocol
+	}{
+		{
+			"nil namedPort policies",
+			[]string{"tcp"},
+			nil,
+			[]string{""},
+		},
+		{
+			"empty map port policies",
+			[]string{"tcp"},
+			make(map[string]*[]NamedNetworkPolicyPort, 0),
+			[]string{""},
+		},
+		{
+			"empty NamedPorts map",
+			[]string{"sctp"},
+			map[string]*[]NamedNetworkPolicyPort{},
+			[]string{""},
+		},
+		{
+			"empty NamedPorts map with no matching pods",
+			[]string{"sctp"},
+			map[string]*[]NamedNetworkPolicyPort{"un-matched-pods": {}},
+			[]string{""},
+		},
+		{
+			"empty NamedPorts map with no matching pods for multiple ports",
+			[]string{"sctp"},
+			map[string]*[]NamedNetworkPolicyPort{
+				"un-matched-port-too-bad": {},
+				"un-matched-port-sad":     {},
+			},
+			[]string{""},
+		},
+		{
+			"single namedPort and single v4 namedPort representation",
+			[]string{"sctp"},
+			map[string]*[]NamedNetworkPolicyPort{
+				"web": {{L4PodPort: "35356", L3PodIPFamily: "ip4", L3PodIP: "10.128.1.4", L4Protocol: "sctp"}},
+			},
+			[]string{"sctp && ((ip4.dst == 10.128.1.4 && sctp.dst == 35356))"},
+		},
+		{
+			"single namedPort and multiple v4 namedPort representations",
+			[]string{"tcp"},
+			map[string]*[]NamedNetworkPolicyPort{
+				"web": {
+					{L4PodPort: "35356", L3PodIPFamily: "ip4", L3PodIP: "10.128.1.4", L4Protocol: "tcp"},
+					{L4PodPort: "35354", L3PodIPFamily: "ip4", L3PodIP: "10.128.1.5", L4Protocol: "tcp"},
+				},
+			},
+			[]string{"tcp && ((ip4.dst == 10.128.1.4 && tcp.dst == 35356) || (ip4.dst == 10.128.1.5 && tcp.dst == 35354))"},
+		},
+		{
+			"single namedPort and multiple v4 & v6 namedPort representations",
+			[]string{"tcp", "sctp"},
+			map[string]*[]NamedNetworkPolicyPort{
+				"web": {
+					{L4PodPort: "35356", L3PodIPFamily: "ip4", L3PodIP: "10.128.1.4", L4Protocol: "tcp"},
+					{L4PodPort: "35356", L3PodIPFamily: "ip6", L3PodIP: "fe00:10:128:1::4", L4Protocol: "tcp"},
+					{L4PodPort: "35354", L3PodIPFamily: "ip4", L3PodIP: "10.128.1.5", L4Protocol: "sctp"},
+					{L4PodPort: "35354", L3PodIPFamily: "ip6", L3PodIP: "fe00:10:128:1::5", L4Protocol: "sctp"},
+				},
+			},
+			[]string{"tcp && ((ip4.dst == 10.128.1.4 && tcp.dst == 35356) || (ip6.dst == fe00:10:128:1::4 && tcp.dst == 35356))",
```

Comment:
@dceara @numansiddique I always forget but in OVN do we have an issue in combining v4 and v6 matches like this for ACLs.?

Reply:
There are cases in which we have issues with that.  In this case I think it's fine but I'd avoid combining them just to be sure.